attendify.js

```diff
@@ -0,0 +1,438 @@
+      return {
+        eventId: curentEventId,
+        account: vaultWallet.classicAddress,
+        owner: walletAddress,
+        URI: url,
+        title: title,
+        claimable: nftokenCount,
+      };
+    } catch (error) {
+      console.error(error);
+      throw new Error(error);
+      // return error;
+    }
+  }
+
+  /**
+   * Verifies whether or not walletAddress account is owner of NFT from event with eventId that was issued by wallet that matches minter parameter. This step only works if user has interacted with the app previously by generating unique id that has to be included in the signed tx memo and turned into hex value
+   * * Wallet from signature has to match walletAddress
+   * @param {string} walletAddress - Address of wallet for the user wanting to verify
+   * @param {string} signature - Signature that should be signed by the same account as walletAddress. This could be done either using XUMM or `sign` function from xrpl library. The mock transaction from signature has to contain memo with number generated for walletAddress. See test.js for example implementation of this
+   * @param {string} minter - The address of the wallet that minted the NFT
+   * @param {number} eventId - The event ID of the NFT
+   * @returns {boolean} Indicats whether the walletAddress owns any NFT from particular event
+   */
+  async verifyOwnership(walletAddress, signature, minter, eventId) {
+    try {
+      if (!walletAddress || !signature || !minter || !eventId)
+        throw new Error(`${ERR_PARAMS}`);
+      const verifySignatureResult = verifySignature(signature);
+      const DECODED_SIGNATURE = xrpl.decode(signature);
+      if (verifySignatureResult.signatureValid != true)
+        throw new Error(`Signature is not valid.`);
+      if (verifySignatureResult.signedBy != walletAddress)
+        throw new Error(
+          `Signature wasn't signed by provided wallet address '${walletAddress}'. Please provide correct signature and try again.`
+        );
+      if (!DECODED_SIGNATURE.Memos)
+        throw new Error(
+          `Signed tx must include memo with ID obtained at the start of verification process.`
+        );
+      const TX_MEMO = xrpl.convertHexToString(
+        DECODED_SIGNATURE.Memos[0].Memo.MemoData
+      );
+      const EXPECTED_MEMO_ID = this.signatureMap.get(walletAddress);
+      if (!EXPECTED_MEMO_ID)
+        throw new Error(
+          `Wallet address '${walletAddress}' don't have the verification ID generated for it yet. Please start verification process by obtaining a verification ID before performing ownership verification check.`
+        );
+      if (TX_MEMO != EXPECTED_MEMO_ID)
+        throw new Error(
+          `Memo '${TX_MEMO}' from signature does not match expected ID for the provided wallet address '${walletAddress}'.`
+        );
+      // Getting user NFTs and checking whether any NFT was issued by minter address
+      const accountNfts = await await this.getBatchNFTokens(
```

Comment:
You're calling await twice
```suggestion
      const accountNfts = await this.getBatchNFTokens(
```